### PR TITLE
fix inquirer types

### PR DESCRIPTION
**Pull request by regevbr**
I added the missing `askAnswered` property on the question interface (https://github.com/SBoudrias/Inquirer.js#question )
I fixed the PromptUI `process` observable type to the real one (https://github.com/SBoudrias/Inquirer.js/blob/a1da6cfc97fe29cdeb1edc8b2aafc20eeaf85dca/packages/inquirer/lib/ui/prompt.js#L105)
I also changed the PromptUI class to be generic as to the `Answers` type

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test inquirer`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/SBoudrias/Inquirer.js#question https://github.com/SBoudrias/Inquirer.js/blob/a1da6cfc97fe29cdeb1edc8b2aafc20eeaf85dca/packages/inquirer/lib/ui/prompt.js#L105
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

### CODE REVIEW

File: types/inquirer/inquirer-tests.ts [hunk 1]
```diff
@@ -172,3 +177,37 @@ fetchAsyncQuestionProperty(
...
+            // DO NOTHING
+        },
+    });
+    // $ExpectError
+    promptResult.ui.process.subscribe({
+        next: (value: {name: string, answer: number}) => {
+            // DO NOTHING
+        },
+    });
+    prompts.complete();
```
Comment (manuth) [inline types/inquirer/inquirer-tests.ts:212]: Your current tests don't check whether the `QuestionAnswer` type is actually working.
Therefore I'd recommend you to add following test:
```suggestion

    promptResult.ui.process.subscribe({
        next: (value) => {
            if (value.name === "str") {
                // $ExpectType string
                value.answer;
            }
            else {
                // $ExpectType number
                value.answer;
            }
        },
    });

    prompts.complete();
```

**Review by neurolag — CHANGES_REQUESTED** [review]
Awesome - looks perfectly fine to me!
I left one more recommendation

File: types/inquirer/inquirer-tests.ts (hunk 1 above)
Comment (regevbr) [reply types/inquirer/inquirer-tests.ts:212]: @manuth Thanks for reviewing, this is actually being tested in line 202 but another test will not hurt, will add it in a second

Comment (manuth) [reply types/inquirer/inquirer-tests.ts:212]: Oooh fair - sorry, didn't catch that one

**Review by neurolag — APPROVED** [review]
Looks fine to me!

**Review by gabritto — APPROVED** [review]

**Pull request merged**